src/login/user-runtime-dir.c

```diff
@@ -165,7 +165,7 @@ static int do_mount(const char *user) {
 }
 
 static int do_umount(const char *user) {
-        char runtime_path[sizeof("/run/user") + DECIMAL_STR_MAX(uid_t)];
+        char runtime_path[STRLEN("/run/user") + DECIMAL_STR_MAX(uid_t)];
```

Comment:
so this shortens the buffer by one. Which is actually ok, since DECIMAL_STR_MAX() actually includes space for a trailing NUL byte, so we should be OK.

Except of course that we aren't, because the expression here is actually wrong, it misses space for the `/` before the UID... (same above...)

Reply:
years ago we had a discussion about DECIMAL_STR_MAX() and most of use agreed that the fact that it includes space for a NUL byte was a mistake and we should instead expect the caller to always include it. We all were too lazy to patch through the codebase for this however, so the macro was never changed. But we did start always addingthe +1 as protection for that day when we actually change the macro...

not sure what the lesson of that is, maybe to add an extra +1 here too?

Reply:
Hmm, right... We had such discussion again in https://github.com/systemd/systemd/pull/26826#discussion_r1545781593.

I still think having `DECIMAL_STR_MAX` contain the size for NUL byte is actually good, for reasons like https://github.com/systemd/systemd/pull/26826#discussion_r1551472139. And wasting one byte really doesn't matter much. The intention of the commit is to make things more readable and consistent with the rest of the codebase, rather than saving one byte...